### ISSUE TITLE
Graphic tool to write/edit chord sequences

### ISSUE THREAD

**Issue by elieserdejesus** [elieserdejesus/JamTaba#801]
Write chords in chat works, but is something very primitive. Let's create something more cool and modern to create chord sequences. Off course is still necessary send a "text" version of chords in chat to be compatible with other ninjam clients.

**Comment by jonjamcam** [comment]
Here's what I've been thinking about for some time how it could work (for me):

IDEA 1:

One implementation I find very well done is the plugin **improvisator**. It's got a graphical view of chords arranged according to scale degree: 

![image](https://cloud.githubusercontent.com/assets/15310433/24619585/d221ecb2-1870-11e7-811e-ba19718e7a98.png)

 Above you can see there are many ways to modify a chord, so you could have in Jamtaba buttons for 7th, 9th 13th, etc 

What I'm thinking is some screen where you can click the chords instead of writing them. It could be with drag and drop too into a timeline if possible. For example the looper has a timeline of this kind:

![image](https://cloud.githubusercontent.com/assets/15310433/24619793/a373536e-1871-11e7-9c59-b6b801786b25.png)

So in short:

![image](https://cloud.githubusercontent.com/assets/15310433/24619906/11d2dcda-1872-11e7-9f73-00ecf48944f0.png)

After you got all your chords in place, you push a button and it sends it in text to other ninjamers.

IDEA 2:

I'm convinced that ninjam could work much better if people could play sometimes using scales and modes, not just progressions of chords. In my opinion the way BPI and BPM works is ideal to create music based on modal scales and have no problems with "wrong" notes. In fact these wrong notes if used wisely can be very nice. 

I don't have an idea at the moment of how to implement this, though, but I'll think about it!

My 2 Cents :smile:

**Comment by elieserdejesus** [comment]
Nice @jonjamcam , I like the Improvator GUI too. 

Your "time line" idea is nice, I have the same ideas about "how organize the chord progression".

I think the Improvisator organization (Degrees, Tonic, Dominant and subdominant derivatives) it's ok for a general harmony plugin, but is too complex **for ninjam context** in my opinion. My idea is work in something like "chords for dummies" :) The Improvisator approach is for advanced and skilled enough musicians only, people with less knowledge in harmony will be very confused.

**Comment by qulf** [comment]
for me idea is ok. many time i have problems witch writing chords. now is not user friendly.

**Comment by qulf** [comment]
looks for that:
https://www.youtube.com/watch?v=uDkFDcJfpXM

**Comment by qulf** [comment]
i use ireal on android and i like them.is nice too see chords when playing

**Comment by elieserdejesus** [comment]
just to remember, this issue is related with #504

**Comment by Hitrate** [comment]
I love how Jamtaba remembers the levels settings of the players, even if you've left the room ?!!! And how it has neat chat features where you no longer have to write the bloody !vote bpi 16 you can just press the button to vote :D So maybe if possible, put a button for each normal mode like:

Button: !vote bpi 2
Button: !vote bpi 4
Button: !vote bpi 8
Button: !vote bpi 12
Button: !vote bpi 16
Button: !vote bpi 24
Button: !vote bpi 32
Button: !vote bpi 48
Button: !vote bpi 64

**Comment by Hitrate** [comment]
Just thought of something else. What if the bloody thing was midi controllable? then you could map your tap tempo pedal to the bpm button, and it could set the time for the room without having to touch the keyboard or mouse, or even look at the screen :D same with levels/faders etc...

**Comment by elieserdejesus** [comment]
@Hitrate , the "midi controllable" idea is discussed in #13 and #169, but these issues are frozen at moment.

The "buttons idea" is interesting too, but I think is better create a new issue to help the organization. At moment we have a lot of ideas in these issues and is easy blow my mind reading all details :)

**Comment by elieserdejesus** [comment]
moved to wishlist https://github.com/elieserdejesus/JamTaba/issues/907
